pkg/handlers/deploy.go

```diff
@@ -159,11 +165,18 @@ func (h *Handler) UpdateDeployResult(w http.ResponseWriter, r *http.Request) {
 
 	alreadySuccessful, err := store.GetStore().IsDownstreamDeploySuccessful(updateDeployResultRequest.AppID, clusterID, currentSequence)
 	if err != nil {
-		logger.Error(err)
+		logger.Error(errors.Wrap(err, "failed to check deploy successful"))
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
 
+	go func() {
+		err := deleteUnusedImages(updateDeployResultRequest.AppID)
```

Comment:
should this only be called if the deployment was successful?

Reply:
updated, though it's not an error since unsuccessful deployment does not change current version sequence, so it doesn't hurt to run garbage collection.